src/applications/caregivers/components/PreSubmitInfo/SubmitLoadingIndicator.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { connect } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
+
+const SubmitLoadingIndicator = ({ submission }) => {
+  const [isLoading, setLoading] = useState(false);
+  const [timer, setTimer] = useState(0);
+  const [loadingMessage, setLoadingMessage] = useState(null);
+  const hasAttemptedSubmit = submission.hasAttemptedSubmit;
+  const isSubmitPending = submission.status === 'submitPending';
+
+  const getLoadingMessage = useCallback(
+    () => {
+      if (timer >= 0) {
+        setLoadingMessage('Please wait while we process your application.');
+      }
+
+      if (timer >= 30) {
+        setLoadingMessage(
+          'We’re processing your application. This may take up to 2 minutes. Please don’t refresh your browser.',
+        );
+      }
+    },
+    [timer],
+  );
+
+  useEffect(
+    () => {
+      if (hasAttemptedSubmit && isSubmitPending) {
+        setLoading(true);
```

Comment:
does `setLoading` ever have to be set to false at any point or is it always true because the operation completed and the component isn't being rendered anymore?

Reply:
Good question so I added set loading to false but it was creating some bugs and cuasing the seconds to multiply I tested multiple times and saw that loading was set to false on unMount probably from the clearInterval